### PR TITLE
#11963 compile dependencies to serverlibs

### LINKED ISSUE #11963: Avoid packaged jars in apps [3]
### Problem Description

Deploying SORMAS (or starting the Payara Domain with already deployed apps) takes often more than 90s.

#11902 showed that the deployment time of `sormas-ear.ear` can be improved by reducing the jars that are bundled in the the ear file.

### Proposed Change

Deploy as much dependencies as possible as serverlibs, don't use scope `compile` in `sormas-backend`, `sormas-ui` and `sormas-rest`

### Added Value/Benefit

Faster deployment time means less downtime when updating SORMAS and less waiting while developing SORMAS.

### Acceptance Criteria

- [x] Keep `sormas-api` as compile dependency.
- [x] All other compile dependencies are resolved as serverlib or was decided to stay as compile dependency.
- [x] Deployment time before and after is measured and documented.

### Implementation Details

1. Check that all current compile dependencies are really used.
    - List swagger dependencies only for `sormas-rest` (check that's not used by other active code).
2. Exclude dependencies that are already provided as glassfish/module. Examples:
    - javax.inject
    - jaxb-api ?
3. Keep Vaadin dependencies inside `sormas-ui`
4. Keep `sormas-api` as compile dependency.

### Mockups

_No response_

### Additional Information

**Caution**: 
1. All developers need to update the libs in the Payara domain (Ant -> `sormas-base/build.xml`-> `deploy-serverlibs`)
2. After this ticket is merged, a standard hotfix procedure is maybe not an option due to a lot of POM changes that might result to pesky merge conflicts.

<details>
<summary>Benchmark</summary>

| Phase           | Run 1         | Run 6      | Run 7       | Comment     |
| ------------- | -----------:| ----------:| ----------:| ------------- |
| 1                  |            41s |            8s |          16s | No log entries |
| 2                  |            13s |          13s |          13s | First log entry until `WELD-000900` | 
| 3                  |            36s |          25s |          24s | End of phase 2 until `sormas-ear was successfully deployed` |
| Success time |            81s |           41s |         42s | Reported in `sormas-ear was successfully deployed` message  |
| Total time     |            90s |           46s |         53s | Total time of all phases   |
| Xmx             |       1024m |       1024m |  1024m |   |
| Deployment | exploded    | autodeploy, no ear libs | autodeploy, reduced ear libs   |  |
| Domain startup |      10s  |           18s | 22s  | More domain libs takes more domain startup time |

</details>

### PR DESCRIPTION
**Pull request by StefanKock**
Fixes #11963

Progress:
- [x] 1. Moved most dependencies in all apps to serverlibs. Exclusions: 
    - `sormas-api`
    - All Vaadin dependencies (`sormas-ui`)
    - `gt-shapefile` (`sormas-backend`), see problem 3
- [x] 2. Don't duplicate glassfish/modules in serverlibs
- [x] 3. Verify deployment of apps in payara domain works
- [x] 4. Measure deployment time (see ticket description, run 7 after the changes)

Problems:
1. I needed to add more test dependencies explicitly for sormas-ui to keep EJB tests running (changing dependencies on sormas-backend from compile to provided meant there were not in the test classpath in sormas-ui any longer).
2. I struggled to move `resteasy` to serverlibs but was able to do it by updating it from 3.15.3 to 4.7.7 and exclude Jakarta APIs that had invalid POMs.
```
sormas-base\build.xml:88: Unable to resolve artifact: Unable to get dependency information: Unable to read the metadata file for artifact 'org.jboss.resteasy:resteasy-client:jar': Invalid JDK version in profile 'compile-java8-release-flag': Unbounded range: [9, for project org.jboss:jboss-parent
  org.jboss.resteasy:resteasy-client:jar:3.15.6.Final
```
3. I failed to move `gt-shapefile` to serverlibs because providing them with Ant fails.
<details>
<summary>Details for gt-shapefile problem</summary>
Exception when running Ant `collect-serverlibs`: `sormas-base\build.xml:88: org.apache.maven.artifact.InvalidArtifactRTException: For artifact {org.osgi:org.osgi.annotation:null:jar}: The version cannot be empty.`

```xml
<!-- trying in sormas-base/pom.xml to avoid the version problem by trying to set it or with exclusions: Did not help.

			<dependency>
				<groupId>org.osgi</groupId>
				<artifactId>org.osgi.annotation</artifactId>
				<version>6.0.0</version>
			</dependency>
			<dependency>
				<groupId>org.osgi</groupId>
				<artifactId>org.osgi.compendium</artifactId>
				<version>5.0.0</version>
			</dependency>
			<dependency>
				<groupId>org.osgi</groupId>
				<artifactId>org.osgi.core</artifactId>
				<version>6.0.0</version>
			</dependency>

			<dependency>
				<groupId>org.geotools</groupId>
				<artifactId>gt-shapefile</artifactId>
				<version>28.2</version>
				<scope>provided</scope>
				<exclusions>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.annotation</artifactId>
					</exclusion>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.compendium</artifactId>
					</exclusion>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.core</artifactId>
					</exclusion>
				</exclusions>
			</dependency>

			<dependency>
				<groupId>systems.uom</groupId>
				<artifactId>systems-common</artifactId>
				<version>2.0.2</version>
				<scope>provided</scope>
				<!-- Used by gt-shapefile -->
				<exclusions>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.annotation</artifactId>
					</exclusion>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.compendium</artifactId>
					</exclusion>
					<exclusion>
						<groupId>org.osgi</groupId>
						<artifactId>org.osgi.core</artifactId>
					</exclusion>
				</exclusions>
			</dependency>
```

</details>

Follow up tickets:
- Avoid usage of commons-collections version 3, switch to collections4 and prevent usage with ArchUnit. commons-collections v3 will remain in classpath as transitive dependency.
- Replace Ant scripts by similarly easy to use Maven means. -> #12074 
- Provide `gt-shapefile` as serverlibs (when Ant is removed).

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

**Comment by StefanKock** [comment]
> Deployment fails with a classloader problem. `smallrye` is only contained in the serverlibs, not in the ear libs. It also happens when I put all smallrye dependencies on compile and remove them in the serverlibs.

Idea: Test as packaged ear, test with IntelliJ, try with the whole 'keycloak-admin-client' dependency stack on 'compile'  scope.

**Comment by MartinWahnschaffe** [comment]
Looked into the keycloak smallrye problem a bit.

The following code is where it fails: https://github.com/payara/Payara/blob/0704b24df099e928276d34e115b8ab4ae492a60a/nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java#LL377C21-L377C33

The SmallryeConfig class does not extend PayaraConfig, as expected here. I'm not sure whether this would be different when including the keycloak-admin-client in the ear.

The main difference I see, that leads to the problem is the update of resteasy from 3.15.3.Final to 4.7.7.Final which introduced smallrye. Later versions for resteasy (5 and 6) don't use smallrye anymore.

**Comment by StefanKock** [comment]
> **UPDATE**: I have tried to update resteasy to 5.0.7.Final. The problem is gone and sormas-ear and sormas-ui sucessfully deploy

I tried it as well and it works, no problem exceptions happened.

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

**Comment by StefanKock** [comment]
I updated the branch to 1.86.0-SNAPSHOT

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

File: sormas-base/pom.xml [hunk 1]
```diff
@@ -445,10 +445,11 @@
 				<scope>provided</scope>
 				<exclusions>
 					<!-- Exclude Payara modules here -->
-					<exclusion>
-						<groupId>net.minidev</groupId>
-						<artifactId>json-smart</artifactId>
-					</exclusion>
+					<!-- json-smart is a server module, but for unknown reason can't be loaded by the json-path domain lib -->
```
Comment (StefanKock) [inline sormas-base/pom.xml:448]: 1. Write: net.minidev:json-smart is a Payara module, but for unknown reason can't be loaded by the json-path domain lib
2. Remove the following commented out code

**Review by StefanKock — COMMENTED** [review]
No commented out code, more precise comment

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12050

**Pull request merged**